test_common/harness/kernelHelpers.cpp

```diff
@@ -809,6 +810,76 @@ int create_single_kernel_helper(cl_context context,
                                 const char *buildOptions,
                                 const bool openclCXX)
 {
+    // For the logic that automactially adds -cl-std it is much cleaner if the
+    // build options have RAII. This buffer will store the potentially updated
+    // build options, in which case buildOptions will point at the string owned
+    // by this buffer.
+    std::string build_options_internal{ buildOptions ? buildOptions : "" };
+
+    // Check the build options for the -cl-std option.
+    if (!buildOptions || !strstr(buildOptions, "-cl-std"))
+    {
+        // If the build option isn't present add it using the latetest OpenCL-C
+        // version supported by the device. This allows calling code to force a
+        // particular CL C version if it is required, but also means that
+        // callers need not specify a version if they want to assume the most
+        // recent CL C.
+
+        // First get all the devices in the context and find the minumum
```

Comment:
Nitpick: this comment isn't _quite_ correct, since it's finding the maximum universally supported OpenCL C version rather than the minimum supported OpenCL C version.

I'm not sure how common this will be, but since this is surprisingly tricky I think it'd be helpful to eventually add a helper or utility function that does this so it could be reused in other places.

Reply:
Good point! I'll update the comment and move this functionality into a standalone function with a declaration in `kernelHelpers.h`, that way it can be reused in other places.